docker/config/config.go

```diff
@@ -55,15 +55,19 @@ func (c *Config) GetCredentials(registry string) (string, string, bool) {
 }
 
 func getAuthJSONString(username, password string) string {
-	if username == "_json_key" {
```

Comment:
I tried to grok this a bit - 005c3ac3610d444e9c180a5c73820ae3c6f08b02 mentions that it was done to fix gcr authentication, but I could not get GCR working end to end until I made these changes. 

It _appeared_ to work initially - list operations against GCR succeeded. I saw subsequent push failures however.